Cubical/Algebra/CommSemiring/Base.agda

```diff
@@ -48,3 +37,54 @@ record CommSemiringStr (A : Type ℓ) : Type (ℓ-suc ℓ) where
 
 CommSemiring : ∀ ℓ → Type (ℓ-suc ℓ)
 CommSemiring ℓ = TypeWithStr ℓ CommSemiringStr
+
+makeIsCommSemiring : {R : Type ℓ} {0r 1r : R} {_+_ _·_ : R → R → R}
+                 (is-setR : isSet R)
+                 (+Assoc : (x y z : R) → x + (y + z) ≡ (x + y) + z)
+                 (+IdR : (x : R) → x + 0r ≡ x)
+                 (+Comm : (x y : R) → x + y ≡ y + x)
+                 (·Assoc : (x y z : R) → x · (y · z) ≡ (x · y) · z)
+                 (·IdR : (x : R) → x · 1r ≡ x)
+                 (·DistR+ : (x y z : R) → x · (y + z) ≡ (x · y) + (x · z))
+                 (AnnihilL : (x : R) → 0r · x ≡ 0r)
+                 (·Comm : (x y : R) → x · y ≡ y · x)
+               → IsCommSemiring 0r 1r _+_ _·_
+makeIsCommSemiring {_+_ = _+_} {_·_ = _·_}
+  is-setR +Assoc +IdR +Comm ·Assoc ·IdR ·DistR+ AnnihilL ·Comm = x
+  where x : IsCommSemiring _ _ _ _
+        IsCommSemiring.isSemiring x =
+          makeIsSemiring
+            is-setR +Assoc +IdR +Comm ·Assoc
+            ·IdR (λ x → ·Comm _ x ∙ (·IdR x))
+            ·DistR+ (λ x y z → (x + y) · z       ≡⟨ ·Comm (x + y) z ⟩
+                               z · (x + y)       ≡⟨ ·DistR+ z x y ⟩
+                               (z · x) + (z · y) ≡[ i ]⟨ (·Comm z x i + ·Comm z y i) ⟩
+                               (x · z) + (y · z) ∎ )
+            ( λ x → ·Comm x _ ∙ AnnihilL x) AnnihilL
+        IsCommSemiring.·Comm x = ·Comm
+
+makeCommSemiring : (R : Type ℓ)
```

Comment:
There has been a motion to remove `make...` and only keep the `makeIs...` for all the algebraic structures:
https://github.com/agda/cubical/issues/598

I guess the discussion has remained inconclusive, but maybe something to keep in mind...

Reply:
I think Anders' arguments do not really apply in this case, since there is no difference in readability and I defined ```makeCommSemiring``` a copattern - apart from using ```(R , str)``` - but I think we wan't that one to unfold. Or does it make a difference to use a function defined with a copattern instead of a direct copattern? 
On the other hand, it is really just boilerplate code and the interesting stuff is in ```makeIsCommSemiring``` so I will have a look how things look like, when I remove it.

Reply:
Looks somewhat better - thanks!